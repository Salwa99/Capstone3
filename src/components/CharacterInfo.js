import React from 'react';
import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';

function CharacterInfo() {
  const { charName } = useParams();

  const characters = useSelector((state) => state.characters);
  const character = characters.find((item) => item.name === charName);
  return (
    <div className="mainInfo">
      <ul className="infoContainer">
        {character && (
          <>
            <li>
              <img src={character.image} alt={character.name} />
            </li>
            <div className="charInfo">
              <li>
                Name:
                {character.name}
              </li>
              <li>
                House:
                {character.house}
              </li>
              <li>
                Birth:
                {character.dateOfBirth}
              </li>
              <li>
                Ancestry:
                {character.ancestry}
              </li>
              <li>
                Actor:
                {character.actor}
              </li>
            </div>
          </>
        )}
        {!character && <p>No Character is found!!</p>}
      </ul>
    </div>
  );
}

export default CharacterInfo;
