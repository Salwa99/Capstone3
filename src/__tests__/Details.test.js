import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import CharacterInfo from '../components/CharacterInfo';
import store from '../redux/ConfigueStore';

it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}><CharacterInfo /></Provider>,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
