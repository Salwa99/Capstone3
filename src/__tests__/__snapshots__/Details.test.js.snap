// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="mainInfo"
>
  <ul
    className="infoContainer"
  >
    <p>
      No Character is found!!
    </p>
  </ul>
</div>
`;
